src/segment-loader.js

```diff
@@ -757,12 +762,11 @@ export default class SegmentLoader extends videojs.EventTarget {
     let currentTime = this.currentTime_();
 
     this.pendingSegment_ = null;
-    // add segment timeline information if we're still using the
-    // same playlist
-    if (segmentInfo && segmentInfo.playlist.uri === this.playlist_.uri) {
-      this.updateTimeline_(segmentInfo);
-      this.trigger('progress');
-    }
+
+    // add segment metadata if it we have gained information during the
+    // last append
+    this.updateTimeline_(segmentInfo);
+    this.trigger('progress');
```

Comment:
`progress` in this file was written in the spirit of the HTML5 `progress` event, which is supposed to fire every ~350ms when media data is being loaded. I think it's fair to count segment loads against an old rendition as progress based on that.